Modes testing (#27)
mvllow/modes.nvim: tests/README.md

# Running Tests

> To run tests, refer to the `Makefile` in the root directory of this project. Once there, run `make test` to begin tests.

Reviewer: ```suggestion
```

```sh
make test
```

If there are issues with dependencies then a clean operation can be ran:

```sh
make clean
```

# Creating Tests

> All tests should be within the `tests/` directory and have `spec` as the last part of the filename, so for example: `my_test_spec.lua`. It will then be automatically added to the tests.

Reviewer: ```suggestion
Add new test files to `tests/`, ensuring the filename ends with `spec`. E.g.  `my_test_spec.lua`. This new file will automatically be added to the tests.
```

> 
> This even works recursively in subdirectories within the `tests/` directory so long as the files are suffixed `spec` like above.

Reviewer: ```suggestion
```

> A very small example test:

Reviewer: ```suggestion
Example test:
```

```lua
describe('small test', function()
	describe('compare text', function()

> 		it('it should return true that the the text is the same', function()

Reviewer: ```suggestion
		it('should return true that the the text is the same', function()
```

remote: Total 170 (delta 6), reused 129 (delta 4), pack-reused 0
Receiving objects: 100% (170/170), 146.36 KiB | 269.00 KiB/s, done.
Resolving deltas: 100% (6/6), done.
git clone --depth=1 https://github.com/mvllow/modes.nvim.git dependencies/pack/vendor/start/modes.nvim
Cloning into 'dependencies/pack/vendor/start/modes.nvim'...
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 9 (delta 0), reused 6 (delta 0), pack-reused 0
Receiving objects: 100% (9/9), done.
nvim --headless --noplugin -u tests/minimal.vim -c "call Test()"
Starting...Scheduling: ./tests/init_spec.lua

========================================
Testing:        .../tests/init_spec.lua
Success ||      small test compare text it should return true that the the text is the same

Success:        1
Failed :        0
Errors :        0
========================================

[Process exited 0]
```

# Adding Dependencies

In the `Makefile` dependencies can be added in the `install_dependencies` section. For instance, if there is a new plugin dependency that needs to be added we can clone it into the dependencies directory.

> 
> For example, if [nui.nvim](https://github.com/MunifTanjim/nui.nvim) became a dependency we can add it like so:

Reviewer: ```suggestion
```
